tests/unit/tensor.cpp

```diff
@@ -0,0 +1,2214 @@
+    RETURN_IF_SKIP(Init());
+
+    auto desc = default_desc();
+    auto info = default_create_info(&desc);
+    VkTensorARM tensor;
+    vk::CreateTensorARM(*m_device, &info, nullptr, &tensor);
+    m_errorMonitor->SetDesiredError("VUID-vkDestroyTensorARM-tensor-09732");
+    vk::DestroyTensorARM(*m_device, tensor, vkt::default_allocator());
+    m_errorMonitor->VerifyFound();
+    // Clean up by correctly destroying the tensor
+    vk::DestroyTensorARM(*m_device, tensor, nullptr);
+}
+
+TEST_F(NegativeTensor, DestroyTensorViewInUse) {
+    TEST_DESCRIPTION("Test destroying a tensor view while it is in use");
+    SetTargetApiVersion(VK_API_VERSION_1_4);
+    AddRequiredExtensions(VK_ARM_TENSORS_EXTENSION_NAME);
+    AddRequiredFeature(vkt::Feature::shaderTensorAccess);
+    AddRequiredFeature(vkt::Feature::timelineSemaphore);
+    AddRequiredFeature(vkt::Feature::tensors);
+    RETURN_IF_SKIP(Init());
+
+    vkt::Tensor tensor(*m_device);
+    tensor.bind_to_mem();
+
+    VkTensorViewCreateInfoARM tensor_view_create_info = vku::InitStructHelper();
+    tensor_view_create_info.tensor = tensor.handle();
+    tensor_view_create_info.format = tensor.format();
+    vkt::TensorView view(*m_device, tensor_view_create_info);
+
+    vkt::Buffer buffer(*m_device, tensor.get_memory_reqs().memoryRequirements.size, VK_BUFFER_USAGE_STORAGE_BUFFER_BIT);
+
+    char const *cs_source = R"glsl(
+        #version 450
+        #extension GL_ARM_tensors : require
+        #extension GL_EXT_shader_explicit_arithmetic_types : require
+        layout(local_size_x = 1, local_size_y = 1, local_size_z = 1) in;
+        layout(set=0, binding=0) uniform tensorARM<int32_t, 1> tens;
+        layout(set=0, binding=1, std430) buffer asd {
+            int32_t out_data[];
+        };
+        void main()
+        {
+            const uint size_x = tensorSizeARM(tens, 0);
+            const uint x = gl_GlobalInvocationID.x % size_x;
+            const uint out_index = gl_GlobalInvocationID.x;
+
+            tensorReadARM(tens, uint[](x), out_data[out_index]);
+        }
+)glsl";
+
+    CreateComputePipelineHelper pipe(*m_device);
+    pipe.cs_ = VkShaderObj::CreateFromGLSL(this, cs_source, VK_SHADER_STAGE_COMPUTE_BIT);
+
+    std::vector<VkDescriptorSetLayoutBinding> bindings = {
+        {0, VK_DESCRIPTOR_TYPE_TENSOR_ARM, 1, VK_SHADER_STAGE_COMPUTE_BIT, nullptr},
+        {1, VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, 1, VK_SHADER_STAGE_COMPUTE_BIT, nullptr}};
+
+    pipe.dsl_bindings_.resize(bindings.size());
+    memcpy(pipe.dsl_bindings_.data(), bindings.data(), bindings.size() * sizeof(VkDescriptorSetLayoutBinding));
+    pipe.CreateComputePipeline();
+    pipe.descriptor_set_.WriteDescriptorTensorInfo(0, &view.handle());
+    pipe.descriptor_set_.WriteDescriptorBufferInfo(1, buffer, 0, VK_WHOLE_SIZE, VK_DESCRIPTOR_TYPE_STORAGE_BUFFER);
+    pipe.descriptor_set_.UpdateDescriptorSets();
+
+    vkt::CommandBuffer cmd(*m_device, m_command_pool);
+    cmd.Begin();
```

Comment:
just use `m_command_buffer` unless you have a reason to make a new command buffer object
```suggestion
    m_command_buffer.Begin();
```